pandas/_libs/lib.pyx

```diff
@@ -1475,11 +1482,17 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
         bint seen_val = False
         flatiter it
 
+    if not util.is_array(value):
+        if isinstance(value, (ABCSeries, ABCExtensionArray, ABCPandasArray)):
```

Comment:
any way to do this without relying on the ABCFoos?

Reply:
I agree it's not ideal, but it  was the best solution I could find that keeps the performance:
* Importing `Series`, `Index` etc. into the global namespace is not possible because of circular import issues
* Importing `Series`, `Index` etc. inside `infer_dtype` is possible, but costs 300 ns per imported class for each function call, so it adds up to something very noticable
* I thought of checking class names as a guard before importing inside the function, but that gives problems with subclasses.

The advantage of importing ABCFoo into the global namespace is that it's a onetime import, so no cost for each function call and it's a nice speed boost. But I also see the problem of importing from pythonland.

IMO the speedup is worth this, but I can also see it's not ideal, so no biggie if it doesn't get accepted.

Reply:
definitely not suggesting using the non-ABC versions.  my thought is that without adding this here, in the `elif hasattr(value, "dtype")` block we could do the _try_infer_map(value.dtype) check before the "Unwrap Series/Index" bit